contracts/src/7683/T1ERC7683.sol

```diff
@@ -127,7 +150,25 @@ contract T1ERC7683 is BasicSwap7683, OwnableUpgradeable, PausableUpgradeable {
     /// @param destinationDomain The domain of the destination chain
     /// @param orderId The ID of the order to verify
     /// @return requestId The ID of the read request
-    function verifySettlement(uint32 destinationDomain, bytes32 orderId) external returns (bytes32 requestId) {
+    function verifySettlement(uint32 destinationDomain, bytes32 orderId) external payable returns (bytes32 requestId) {
+        requestId = _verifyFill(destinationDomain, orderId);
+        emit SettlementVerificationRequested(orderId, requestId);
+    }
+
+    function verifyRefund(uint32 destinationDomain, bytes32 orderId) external returns (bytes32 requestId) {
```

Comment:
This is a potential DoS vector. Anyone can call `verifyRefund`, which will set `orderStatus[orderId] = REFUND_REQUESTED;`. After that it would be impossible to call `verifySettlement` because `orderStatus[orderId] != OPENED`, and there would be no way to change the status back to `OPENED`. This would permanently lock all user escrowed funds in the contract.